cli/pom.xml

```diff
@@ -140,54 +140,141 @@
+                <id>build-native</id>
+                <goals>
+                  <goal>compile-no-fork</goal>
+                </goals>
                 <phase>package</phase>
+              </execution>
+              <execution>
+                <id>test-native</id>
                 <goals>
-                  <goal>attach-artifact</goal>
+                  <goal>test</goal>
+                </goals>
+                <phase>test</phase>
+              </execution>
+            </executions>
+            <configuration>
+              <metadataRepository>
+                <enabled>true</enabled>
+              </metadataRepository>
+              <agent>
+                <enabled>true</enabled>
+              </agent>
+              <imageName>${imageName}</imageName>
+              <buildArgs>
+                <arg>--enable-url-protocols=http,https</arg>
+                <arg>-H:IncludeResources="nls/.*"</arg>
+                <arg>--initialize-at-build-time=org.apache.commons.compress</arg>
+              </buildArgs>
+            </configuration>
+          </plugin>
+          <!-- Assembly Plugin -->
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-assembly-plugin</artifactId>
+            <version>${assembly.maven.plugin.version}</version>
+            <executions>
+              <execution>
+                <id>create-distribution</id>
+                <phase>package</phase>
+                <goals>
+                  <goal>single</goal>
                 </goals>
                 <configuration>
-                  <artifacts>
-                    <artifact>
-                      <file>${project.build.directory}/${project.artifactId}-linux-x86_64.tar.gz</file>
-                      <type>tar.gz</type>
-                      <classifier>linux-x64</classifier>
-                    </artifact>
-                    <artifact>
-                      <file>${project.build.directory}/${project.artifactId}-windows-x86_64.tar.gz</file>
-                      <type>tar.gz</type>
-                      <classifier>windows-x64</classifier>
-                    </artifact>
-                    <artifact>
-                      <file>${project.build.directory}/${project.artifactId}-osx-x86_64.tar.gz</file>
-                      <type>tar.gz</type>
-                      <classifier>mac-x64</classifier>
-                    </artifact>
-                    <artifact>
-                      <file>${project.build.directory}/${project.artifactId}-osx-aarch_64.tar.gz</file>
-                      <type>tar.gz</type>
-                      <classifier>mac-arm</classifier>
-                    </artifact>
-                  </artifacts>
+                  <appendAssemblyId>true</appendAssemblyId>
+                  <descriptors>
+                    <descriptor>/src/main/assembly/release-windows-x86_64.xml</descriptor>
+                  </descriptors>
+                </configuration>
```

Comment:
As described above. Just keep one of these 4 profiles as `native` and simply reference descriptor as
```
<descriptor>/src/main/assembly/release-${ideasy.assembly.id}.xml</descriptor>
```
Rename the descriptors to match the github OS image names and we can avoid all the copy & paste.

Reply:
I've implemented all of your suggestions now and tested the release and nightly build workflows on my own fork.
Seems to be working as expected.